pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -642,12 +642,44 @@ func (a *ApplicationBackupController) handle(ctx context.Context, backup *stork_
 	case stork_api.ApplicationBackupStageApplications:
 		err := a.backupResources(backup)
 		if err != nil {
+			isLargeResourceError, err := utils.CheckAndReorganizeError(err)
 			message := fmt.Sprintf("Error backing up resources: %v", err)
 			log.ApplicationBackupLog(backup).Errorf(message)
 			a.recorder.Event(backup,
 				v1.EventTypeWarning,
 				string(stork_api.ApplicationBackupStatusFailed),
 				message)
+			// Check if the error is due to a large resource issue
```

Comment:
Same code two places will it be good to make it function ?  this path will look neat.